runtime/lua/vim/vimhelp.lua

```diff
@@ -2,26 +2,26 @@
 
 local M = {}
 
--- Called when editing the doc/syntax.txt file
-function M.highlight_groups()
-  local save_cursor = vim.fn.getcurpos()
-
-  local start_lnum = vim.fn.search([[\*highlight-groups\*]], 'c')
-  if start_lnum == 0 then
-    return
-  end
-  local end_lnum = vim.fn.search('^======')
-  if end_lnum == 0 then
-    return
-  end
-
+-- apply current colorscheme to lists of default highlight groups
+---@param patterns table[] {start, stop, match} assumed to be sorted by start!
```

Comment:
I believe new types should be as strict as possible. That is, the type of `patterns` should specify the keys and respective value types. This yields better completion and diagnostic support.

Reply:
As this is not a proper AP (and certainly not meant to be used publicly), I didn't even plan on adding any... Feel free to suggest a proper type annotation, and I'll include it.

Reply:
But you're right, the annotation should be correct or not at all. I've removed it for now; if someone wants to promote it to a public API in a followup, a proper docstring (and validation, and whatnot) can be added.

Reply:
```lua
--- @param patterns {start:string,stop:string,match:string}[]
```

Reply:
I feel like proper types aren't just a requirement for public APIs, they're also for our own developing enjoyment. Better completions and diagnostics!!

Reply:
That's why I put them in.

Reply:
Right, but I just wanted to make sure you didn't grumpily added it but were actually convinced of the change ;)

Reply:
No, I did add it grumpily :) But it's the result that counts.